openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java

```diff
@@ -170,8 +172,14 @@ null, createIdentifierToken(GeneratorConstants.BYTE)),
                             GeneratorUtils.getValidName(schemaName, true))));
                     break;
                 default:
-                    identifierToken = createIdentifierToken(GeneratorConstants.JSON);
-                    typeName = Optional.ofNullable(createSimpleNameReferenceNode(identifierToken));
+                    ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitionNode>> mediaTypeTokens =
+                            handleMediaType(mediaType, null);
+                    if (mediaTypeTokens.left.isEmpty()) {
+                        identifierToken = createIdentifierToken(HTTP_REQUEST);
+                        typeName = Optional.ofNullable(createSimpleNameReferenceNode(identifierToken));
+                    } else {
+                        typeName = mediaTypeTokens.left;
```

Comment:
Shall we swap the if else order in here, as we usually handle the happy path within the if case?

Reply:
Updated